tracker/enqueue.go

```diff
@@ -61,6 +62,24 @@ type set map[string]time.Time
 
 // Track implements Interface.
 func (i *impl) Track(ref corev1.ObjectReference, obj interface{}) error {
+	var missingFields []string
+	if ref.APIVersion == "" {
+		missingFields = append(missingFields, "APIVersion")
+	}
+	if ref.Kind == "" {
+		missingFields = append(missingFields, "Kind")
+	}
+	if ref.Namespace == "" {
+		missingFields = append(missingFields, "Namespace")
+	}
+	if ref.Name == "" {
```

Comment:
I recommend testing like `if errs := validation.IsQualifiedName(ref.Name); len(errs) > 0 {` https://github.com/knative/eventing/blob/56618d44cfa807ba9a6d7ed26faece5c91447e65/pkg/apis/feeds/v1alpha1/feed_validation.go#L45

Reply:
While we place this limitation on Knative resources, this is not a general requirement in K8s, and so we could be asked to track a resource with a name that departs from this requirement.

Reply:
I think you'd want `IsDNS1123Label` rather than `IsQualifiedName`; the latter accepts strings like `foo.example.com/Bar`